pymc_marketing/clv/distributions.py

```diff
@@ -460,3 +459,221 @@ def logp(value, r, alpha, s, beta, T):
+        [(\frac{s}{r+s+x})_2F_1(r+s+x,s+1;r+s+x+1;\frac{\alpha-\beta}{\alpha+t_x}) \\
+        &+ (\frac{r+x}{r+s+x})
+        \frac{_2F_1(r+s+x,s;r+s+x+1;\frac{\alpha-\beta}{\alpha+T})(\alpha +t_x)^{r+s+x}}
+        {(\alpha +T)^{r+s+x}}] \\
+        \\
+        \text{if }\beta >= \alpha: \\
+        \\
+        \mathbb{L}(r, \alpha, s, \beta | x, t_x, T) &=
+        \frac{\Gamma(r+x)\alpha^r\beta}{\Gamma(r)+(\beta +t_x)^{r+s+x}}
+        [(\frac{s}{r+s+x})_2F_1(r+s+x,r+x;r+s+x+1;\frac{\beta-\alpha}{\beta+t_x}) \\
+        &+ (\frac{r+x}{r+s+x})
+        \frac{_2F_1(r+s+x,r+x+1;r+s+x+1;\frac{\beta-\alpha}{\beta+T})(\beta +t_x)^{r+s+x}}
+        {(\beta +T)^{r+s+x}}]
+        \end{align}
+
+    ========  ===============================================
+    Support   :math:`t_j > 0` for :math:`j = 1, \dots, x`
+    Mean      :math:`\mathbb{E}[X(t) | r, \alpha, s, \beta] = \frac{r\beta}{\alpha(s-1)}[1-(\frac{\beta}{\beta + t})^{s-1}]`
+    ========  ===============================================
+
+    References
+    ----------
+    .. [1] Fader, Peter S., Bruce G.S. Hardie, and Jen Shang (2010),
+       "Customer-Base Analysis in a Discrete-Time Noncontractual Setting,"
+       Marketing Science, 29 (6), 1086-1108. https://www.brucehardie.com/papers/020/fader_et_al_mksc_10.pdf
+
+    """
+
+    rv_op = beta_geo_beta_binom
+
+    @classmethod
+    def dist(cls, alpha, beta, gamma, delta, T, **kwargs):
+        return super().dist([alpha, beta, gamma, delta, T], **kwargs)
+
+    def logp(value, alpha, beta, gamma, delta, T):
+        t_x = pt.atleast_1d(value[..., 0])
+        x = pt.atleast_1d(value[..., 1])
+        scalar_case = t_x.type.broadcastable == (True,)
+
+        for param in (t_x, x, alpha, beta, gamma, delta, T):
+            if param.type.ndim > 1:
+                raise NotImplementedError(
+                    f"BetaGeoBetaBinom logp only implemented for vector parameters, got ndim={param.type.ndim}"
+                )
+            if scalar_case:
+                if param.type.broadcastable == (False,):
+                    raise NotImplementedError(
+                        f"Parameter {param} cannot be larger than scalar value"
+                    )
+
+        # Broadcast all the parameters so they are sequences.
+        # This is potentially inefficient, but otherwise we need some ugly logic to unpack argumens in the scan function,
+        # since sequences always precede non-sequences.
+        _, alpha, beta, gamma, delta, T = pt.broadcast_arrays(
+            t_x, alpha, beta, gamma, delta, T
+        )
+
+        def logp_customer_died(t_x_i, x_i, alpha_i, beta_i, gamma_i, delta_i, T_i):
+            i = pt.scalar("i", dtype=int)
+            died = pt.lt(t_x_i + i, T_i)
+
+            unnorm_logprob_customer_died_at_tx_plus_i = betaln(
+                alpha_i + x_i, beta_i + t_x_i - x_i + i
+            ) + betaln(gamma_i + died, delta_i + t_x_i + i)
```

Comment:
The only difference between the series in the sum and the ones outside, in the paper and lifetimes), is that +1 after the gamma. By making it a boolean variable we can handle everything in the same sum.